### PR TITLE
Added an option to generate an AES-128 key and related unit tests

### PR DESCRIPTION
**Pull request by Songroid**
We are using `Google-java-format` plugin (AOSP style) in Android Studio. That's why there's so many format changes... Any suggestion to improve this?

### CODE REVIEW

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java [hunk 1]
```diff
@@ -67,14 +67,25 @@
...
 
     /**
-     * Returns the unique ID being used.
+     * Returns the unique ID being used. The default key length is 256 bits.
      *
      * @param name Unique name associated with this unique ID.
      * @return Unique ID.
      */
     public static synchronized String getUniqueId(String name) {
+        return getUniqueId(name, 256);
```
Comment (Songroid) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java]: Backward compatibility. The default option is 256 bits.

**Comment by bhariharan** [comment]
@Songroid Could you disable the plugin and remove the formatting changes please?

File: libs/test/SalesforceSDKTest/src/com/salesforce/androidsdk/app/SalesforceKeyGeneratorTest.java [hunk 2]
```diff
@@ -52,24 +52,35 @@
...
+    @Test
+    public void testGetUniqueId() {
+        final String id1 = SalesforceKeyGenerator.getUniqueId(KEY_1);
         final String id1Again = SalesforceKeyGenerator.getUniqueId(KEY_1);
-		final String id2 = SalesforceKeyGenerator.getUniqueId(KEY_2);
-		Assert.assertEquals("Unique IDs with the same name should be the same", id1, id1Again);
+        final String id2 = SalesforceKeyGenerator.getUniqueId(KEY_2);
+        // 4*Math.Ceiling(((double)bytes.Length/3))) + length of getAddendum(KEY)
+        // 4*Math.Ceiling(32/3)+14 = 58
+        Assert.assertEquals(id1.length(), 58);
```
Comment (Songroid) [inline libs/test/SalesforceSDKTest/src/com/salesforce/androidsdk/app/SalesforceKeyGeneratorTest.java]: Make sure the generated string length matches with the original key size.

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">📱 Needs testing on a real device if change is non-trivial.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_PR-Check">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java [hunk 3]
```diff
@@ -73,8 +73,20 @@
      * @return Unique ID.
      */
     public static synchronized String getUniqueId(String name) {
```
Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java]: Remove `synchronized` keyword from this method and move it to the method below.

Comment (bhariharan) [reply libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java]: Also, could you add a comment in the `Javadoc` for this method that the default key length used is `256`?

File: libs/test/SalesforceSDKTest/src/com/salesforce/androidsdk/app/SalesforceKeyGeneratorTest.java [hunk 4]
```diff
@@ -65,9 +66,18 @@ public void testGetUniqueId() {
 		final String id1 = SalesforceKeyGenerator.getUniqueId(KEY_1);
         final String id1Again = SalesforceKeyGenerator.getUniqueId(KEY_1);
 		final String id2 = SalesforceKeyGenerator.getUniqueId(KEY_2);
+        // Output: 4*Math.Ceiling(((double)bytes.Length/3))) + length of getAddendum(KEY)
+        // 4*Math.Ceiling(32/3)+14 = 58
+        Assert.assertEquals(id1.length(), 58);
```
Comment (bhariharan) [inline libs/test/SalesforceSDKTest/src/com/salesforce/androidsdk/app/SalesforceKeyGeneratorTest.java]: Add a message here if it fails.

File: libs/test/SalesforceSDKTest/src/com/salesforce/androidsdk/app/SalesforceKeyGeneratorTest.java [hunk 5]
```diff
@@ -65,9 +66,18 @@ public void testGetUniqueId() {
...
 		Assert.assertEquals("Unique IDs with the same name should be the same", id1, id1Again);
         Assert.assertNotSame("Unique IDs with different names should be different", id1, id2);
-	}
+
+        final String id3 = SalesforceKeyGenerator.getUniqueId(KEY_3, 128);
+        final String id3Again = SalesforceKeyGenerator.getUniqueId(KEY_3, 128);
+        // 4*Math.Ceiling(16/3)+14 = 38
+        Assert.assertEquals(id3.length(), 38);
```
Comment (bhariharan) [inline libs/test/SalesforceSDKTest/src/com/salesforce/androidsdk/app/SalesforceKeyGeneratorTest.java]: Add a message here if it fails.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java (hunk 3 above)
Comment (Songroid) [reply libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java]: Any specific reason to move it to the inside method? Just curious

**Pull request merged**